mobile/library/common/jni/types/exception.cc

```diff
@@ -0,0 +1,152 @@
+#include "library/common/jni/types/exception.h"
+
+#include "source/common/common/assert.h"
+
+#include "fmt/format.h"
+#include "library/common/jni/types/env.h"
+#include "library/common/jni/types/string.h"
+
+namespace Envoy {
+namespace JNI {
+
+bool Exception::checkAndClear(const std::string& details) {
+  auto env = Env::get();
+  if (env->ExceptionCheck() == JNI_TRUE) {
+    jthrowable throwable = env->ExceptionOccurred();
+    env->ExceptionClear();
+
+    const auto exception = Exception(env, throwable);
+    ENVOY_LOG_EVENT_TO_LOGGER(GET_MISC_LOGGER(), info, "jni_cleared_pending_exception",
+                              exception.description(details));
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * @brief Creates a description of an exception in the following format:
+ * GENERIC_EXCEPTION_DESCRIPTION||DETAIL||EXCEPTION_STACKTRACE||EXCEPTION_CAUSE_STACKTRACE
+ */
+std::string Exception::description(const std::string& detail) const {
+  auto throwable_cause_description = causedByThrowableDescription();
+  std::vector<std::string> descriptionComponents = {
+      throwableDescription(throwable_),
+      detail == "" ? "NO_DETAIL" : detail,
+      throwableStacktraceDescription(throwable_),
+      throwable_cause_description == "" ? "NO_CAUSE" : throwable_cause_description, 
```

Comment:
is NO_CAUSE accurate or is it more UNKNOWN_CAUSE?

Reply:
I found some docs about the `cause` property:

>getCause - Returns the cause of this throwable or null if the cause is nonexistent or unknown. (The cause is the throwable that caused this throwable to get thrown.)

so I think that I will either go with your suggestion `CAUSE_UNKNOWN` or `CAUSE_UNKNOWN_OR_NONEXISTENT`